proposals/dao/tip_118.ts

```diff
@@ -0,0 +1,198 @@
+  logging && console.log('DAI holding deposit deployed to: ', daiHoldingDeposit.address);
+
+  const gOHMHoldingPCVDeposit = await ERC20HoldingPCVDepositFactory.deploy(addresses.core, addresses.gOHM);
+  await gOHMHoldingPCVDeposit.deployTransaction.wait();
+  logging && console.log('gOHM holding deposit deployed to: ', gOHMHoldingPCVDeposit.address);
+
+  return {
+    wethHoldingDeposit,
+    lusdHoldingDeposit,
+    voltHoldingDeposit,
+    daiHoldingDeposit,
+    gOHMHoldingPCVDeposit
+  };
+};
+
+// Do any setup necessary for running the test.
+// This could include setting up Hardhat to impersonate accounts,
+// ensuring contracts have a specific state, etc.
+const setup: SetupUpgradeFunc = async (addresses, oldContracts, contracts, logging) => {
+  initialDAIPSMFeiBalance = await contracts.fei.balanceOf(addresses.daiFixedPricePSM);
+};
+
+// Tears down any changes made in setup() that need to be
+// cleaned up before doing any validation checks.
+const teardown: TeardownUpgradeFunc = async (addresses, oldContracts, contracts, logging) => {
+  console.log(`No actions to complete in teardown for fip${fipNumber}`);
+};
+
+// Run any validations required on the fip using mocha or console logging
+// IE check balances, check state of contracts, etc.
+const validate: ValidateUpgradeFunc = async (addresses, oldContracts, contracts, logging) => {
+  ////////// Validate empty PCV deposit deployments
+  const core = contracts.core;
+  const ethPSM = contracts.ethPSM;
+  const lusdPSM = contracts.lusdPSM;
+  const raiPSM = contracts.raiPriceBoundPSM;
+
+  const fei = contracts.fei;
+  const lusd = contracts.lusd;
+  const weth = contracts.weth;
+  const volt = contracts.volt;
+  const gOHM = contracts.gOHM;
+
+  const wethHoldingDeposit = contracts.wethHoldingDeposit;
+  const lusdHoldingDeposit = contracts.lusdHoldingDeposit;
+  const voltHoldingDeposit = contracts.voltHoldingDeposit;
+  const daiHoldingDeposit = contracts.daiHoldingDeposit;
+  const gOHMHoldingPCVDeposit = contracts.gOHMHoldingPCVDeposit;
+
+  const pcvGuardian = contracts.pcvGuardianNew;
+
+  const EXPECTED_WETH_TRANSFER = toBN('23129630802267046361289');
+  const EXPECTED_LUSD_TRANSFER = toBN('17765325999630072368537481');
+  const EXPECTED_VOLT_TRANSFER = ethers.constants.WeiPerEther.mul(10_000_000);
+
+  // 1. Validate all holding PCV Deposits configured correctly
+  expect(await wethHoldingDeposit.balanceReportedIn()).to.be.equal(addresses.weth);
+  expect(await lusdHoldingDeposit.balanceReportedIn()).to.be.equal(addresses.lusd);
+  expect(await voltHoldingDeposit.balanceReportedIn()).to.be.equal(addresses.volt);
+  expect(await daiHoldingDeposit.balanceReportedIn()).to.be.equal(addresses.dai);
+  expect(await gOHMHoldingPCVDeposit.balanceReportedIn()).to.be.equal(addresses.gOHM);
+
+  // 2. Validate can drop funds on a PCV Deposit and then withdraw with the guardian
+  const wethWhale = '0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2';
```

Comment:
lol the WETH contract is a WETH whale ? sad :-/

Reply:
could you just forceEth(wethHoldingDeposit.address, 'amount') and test wrapETH() at the same occasion ? instead of transferring WETH from outside

Reply:
Nice idea! Will do